### PR TITLE
`CMS_ContentInfo_free()`: fix mem leak on encrypted content key

### LINKED ISSUE #21026: [REGRESSION] CMS_Final() fails when modifying CMS
Root cause: #19919 "Fix SMIME_crlf_copy() to properly report an error"
`master` 9e5bd8923bff3e4f0cbba05c7dadfe289c66eb6f
`>=openssl-3.0.8` 6259cf34eab1abdcfde12b7b3892b4780844091b

Previous behavior required `CMS_final()` to be called when CMS is modified (for example add recepient).
Currently, the `CMS_final()` fails with:

```
802BF10400000000:error:100C0102:BIO routines:bio_read_intern:passed a null parameter:crypto/bio/bio_lib.c:274:
802BF10400000000:error:1C80006B:Provider routines:ossl_cipher_generic_block_final:wrong final block length:providers/implementations/ciphers/ciphercommon.c:429:
```

Interestingly, if the `CMS_final()` is removed, the CMS is created correctly, however, it leaks memory.

Reproduction project is available here[1].

[1] https://github.com/alonbl/openssl-regression-cms

### PR DESCRIPTION
**Pull request by DDvO**
Also check for NULL pointer arguments in `SMIME_crlf_copy()`.

Fixes #21026

### CODE REVIEW

**Comment by alonbl** [comment]
Thank you, I can confirm that it works.

File: crypto/asn1/asn_mime.c [hunk 1]
```diff
@@ -516,6 +516,12 @@ int SMIME_crlf_copy(BIO *in, BIO *out, int flags)
     int len;
     char linebuf[MAX_SMLEN];
     int ret;
+
+    if (in == NULL || out == NULL) {
+        ERR_raise(ERR_LIB_ASN1, ERR_R_PASSED_NULL_PARAMETER);
```
Comment (bernd-edlinger) [inline crypto/asn1/asn_mime.c:521]: BTW: I always wondered why the result of BIO_write/BIO_printf etc. is not checked in this function

**Review by t8m — APPROVED** [review]

File: crypto/asn1/asn_mime.c (hunk 1 above)
Comment (DDvO) [reply crypto/asn1/asn_mime.c:521]: Nice to hear.
Then how about giving your committer's approval?

Comment (bernd-edlinger) [reply crypto/asn1/asn_mime.c:521]: Yes, I'll do that if you add the error handling for the write functions.

**Review by paulidale — APPROVED** [review]

**Pull request closed without merging**